pkg/reconciler/pubsub/pullsubscription/k8s/controller.go

```diff
@@ -74,24 +78,37 @@ func NewController(
 	}
 
 	r := &Reconciler{
-		PubSubBase:             pubsubBase,
-		deploymentLister:       deploymentInformer.Lister(),
-		pullSubscriptionLister: pullSubscriptionInformer.Lister(),
-		receiveAdapterImage:    env.ReceiveAdapter,
-		createClientFn:         gpubsub.NewClient,
+		Base: &psreconciler.Base{
+			PubSubBase:             pubsubBase,
+			DeploymentLister:       deploymentInformer.Lister(),
+			PullSubscriptionLister: pullSubscriptionInformer.Lister(),
+			ReceiveAdapterImage:    env.ReceiveAdapter,
+			CreateClientFn:         gpubsub.NewClient,
+			ControllerAgentName:    controllerAgentName,
+			FinalizerName:          finalizerName,
+		},
 	}
 
 	impl := controller.NewImpl(r, pubsubBase.Logger, reconcilerName)
 
 	pubsubBase.Logger.Info("Setting up event handlers")
-	pullSubscriptionInformer.Informer().AddEventHandlerWithResyncPeriod(controller.HandleAll(impl.Enqueue), reconciler.DefaultResyncPeriod)
+
+	onlyKedaScaler := pkgreconciler.AnnotationFilterFunc(duckv1alpha1.AutoscalingClassAnnotation, duckv1alpha1.KEDA, false)
```

Comment:
Is there any good way to generalize this and or pull it out into a helper somewhere with generic scaling code?  Not worth a lot of engineering effort now since it's unlikely we'll be adding a third option to pubsub anytime soon, but it would be nice if we didn't have one random piece of code to know about/fix that is hard coded to the existence of only a single other option (keda) rather than a potential family of other options.

Reply:
Valid point. Given that is unlikely we add a new strategy anytime soon, I just added a comment in the code. If things play out nice with this, I'll try to upstream some pieces to pkg, and revisit this again. Sounds good?

Reply:
I'm ok with that for now.